### PR TITLE
Moving user search to service

### PR DESCRIPTION
**Pull request by zapu**
in preparation for contact searching

### CODE REVIEW

File: protocol/bin/enabled-calls.json [hunk 1]
```diff
@@ -168,13 +167,11 @@
   "keybase.1.login.paperKey": {"engineSaga": true},
   "keybase.1.login.paperKeySubmit": {"promise": true},
   "keybase.1.notifyCtl.setNotifications": {"promise": true},
-  "keybase.1.phoneNumbers.addPhoneNumber": {"promise": true},
-  "keybase.1.phoneNumbers.deletePhoneNumber": {"promise": true},
-  "keybase.1.phoneNumbers.verifyPhoneNumber": {"promise": true},
-  "keybase.1.phoneNumbers.setVisibilityPhoneNumber": {"promise": true},
```
Comment (zapu) [inline protocol/bin/enabled-calls.json]: there were some duplicates here and other stuff

**Comment by zapu** [comment]
cc @keybase/react-hackers @keybase/y2ksquad

**Comment by zapu** [comment]
@heronhaye can you review go side?

File: shared/actions/team-building.tsx [hunk 2]
```diff
@@ -13,27 +13,21 @@ export type NSAction = {payload: {namespace: TeamBuildingTypes.AllowedNamespace}
 const apiSearch = (
   query: string,
   service: TeamBuildingTypes.ServiceIdWithContact,
-  limit: number,
+  maxResults: number,
```
Comment (buoyad) [inline shared/actions/team-building.tsx]: 👍

File: shared/actions/team-building.tsx [hunk 3]
```diff
@@ -13,27 +13,21 @@ export type NSAction = {payload: {namespace: TeamBuildingTypes.AllowedNamespace}
...
+    maxResults,
+    query,
+    service,
   })
-    .then(results =>
-      JSON.parse(results.body)
-        .list.map(r => Constants.parseRawResultToUser(r, service))
-        .filter(u => !!u)
-    )
+    .then(results => results.map(r => Constants.parseRawResultToUser(r, service)).filter(u => !!u))
```
Comment (buoyad) [inline shared/actions/team-building.tsx]: Switch to `.filter(Boolean)`

File: shared/actions/search.tsx [hunk 4]
```diff
@@ -144,15 +144,17 @@ function _parseSuggestion(username: string, fullname: string) {
...
-function _apiSearch(searchTerm: string, service: string = '', limit: number = 20): Promise<Array<RawResult>> {
-  return RPCTypes.apiserverGetWithSessionRpcPromise({
-    args: [
-      {key: 'q', value: trim(searchTerm)},
-      {key: 'num_wanted', value: String(limit)},
-      {key: 'service', value: service === 'Keybase' ? '' : service},
-    ],
-    endpoint: 'user/user_search',
-  }).then(results => JSON.parse(results.body).list || [])
+function _apiSearch(
```
Comment (buoyad) [inline shared/actions/search.tsx]: 🐑 Rename this, `callSearch`?

**Pull request merged**